src/cdk-experimental/listbox/listbox.ts

```diff
@@ -7,98 +7,206 @@
+
+  activateOption() {
+    if (!this._getComboDisabled()) {
+      this._active = true;
+      this.listbox.setActiveOption(this);
+    }
+  }
+
+  deactivateOption() {
+    this._active = false;
+  }
+
+  _getComboDisabled(): boolean {
+    return (this.listbox.disabled || this._disabled);
+  }
+
+  _getTabIndex(): string | null {
+    return (this.listbox.disabled || this._disabled) ? null : '-1';
+  }
+
+  getLabel(): string {
+    return this._el.nativeElement.textContent;
+  }
+
+  setActiveStyles() {
+    this._active = true;
+  }
+
+  setInactiveStyles() {
+    this._active = false;
   }
 
   static ngAcceptInputType_selected: BooleanInput;
+  static ngAcceptInputType_disabled: BooleanInput;
 }
 
-/**
- * Directive that applies interaction patterns to an element following the aria role of listbox.
- * Typically CdkOption elements are placed inside the listbox. Logic to handle keyboard navigation,
- * selection of options, active options, and disabled states is built in.
- */
 @Directive({
-  selector: '[cdkListbox]',
-  exportAs: 'cdkListbox',
-  host: {
-    role: 'listbox',
-  }
+    selector: '[cdkListbox]',
+    exportAs: 'cdkListbox',
+    host: {
+      role: 'listbox',
+      '(keydown)': '_keydown($event)',
+      '[attr.aria-disabled]': '_disabled',
+    }
 })
-export class CdkListbox {
+export class CdkListbox implements AfterContentInit, OnDestroy {
+
+  _listKeyManager: ActiveDescendantKeyManager<CdkOption>;
+  private _disabled: boolean = false;
 
-  /** A query list containing all CdkOption elements within this listbox */
   @ContentChildren(CdkOption, {descendants: true}) _options: QueryList<CdkOption>;
 
   @Output() readonly selectionChange: EventEmitter<ListboxSelectionChangeEvent> =
       new EventEmitter<ListboxSelectionChangeEvent>();
 
+  @Input()
+  get disabled(): boolean {
+    return this._disabled;
+  }
+  set disabled(value: boolean) {
+    this._disabled = coerceBooleanProperty(value);
+  }
+
+  ngAfterContentInit() {
+    this._listKeyManager = new ActiveDescendantKeyManager(this._options)
+      .withWrap().withVerticalOrientation().withTypeAhead();
+  }
+
+  ngOnDestroy() {
+    this._listKeyManager.change.complete();
+  }
+
+  _keydown(event: KeyboardEvent) {
+    if (this._disabled) {
+      return;
+    }
+
+    const keyCode = event.keyCode;
+
+    switch (keyCode) {
+      case SPACE:
+      case ENTER:
+        if (this._listKeyManager.activeItem && !this._listKeyManager.isTyping()) {
+          this._toggleActiveOption();
+        }
+        break;
+      default:
+        this._listKeyManager.onKeydown(event);
+    }
+  }
+
   /** Emits a selection change event, called when an option has its selected state changed */
   _emitChangeEvent(option: CdkOption) {
     this.selectionChange.emit(new ListboxSelectionChangeEvent(this, option));
   }
 
-  /** Sets the given option's selected state to true */
+  private _toggleActiveOption() {
+    const currentActiveOption = this._listKeyManager.activeItem;
```

Comment:
```suggestion
    const activeOption = this._listKeyManager.activeItem;
```
Not universally true, but the word "current" in variable names tends to be superfluous